File: Python/Package/underautomation/universal_robots/primary_interface/internal/primary_interface_client_base.py
import typing
from underautomation.universal_robots.primary_interface.robot_mode_data_package_event_args import RobotModeDataPackageEventArgs
from underautomation.universal_robots.primary_interface.joint_data_package_event_args import JointDataPackageEventArgs
from underautomation.universal_robots.primary_interface.tool_data_package_event_args import ToolDataPackageEventArgs
from underautomation.universal_robots.primary_interface.masterboard_data_package_event_args import MasterboardDataPackageEventArgs
from underautomation.universal_robots.primary_interface.cartesian_info_package_event_args import CartesianInfoPackageEventArgs
from underautomation.universal_robots.primary_interface.kinematics_info_package_event_args import KinematicsInfoPackageEventArgs
from underautomation.universal_robots.primary_interface.configuration_data_package_event_args import ConfigurationDataPackageEventArgs
from underautomation.universal_robots.primary_interface.force_mode_data_package_event_args import ForceModeDataPackageEventArgs
from underautomation.universal_robots.primary_interface.additional_info_package_event_args import AdditionalInfoPackageEventArgs
from underautomation.universal_robots.primary_interface.calibration_data_package_event_args import CalibrationDataPackageEventArgs
from underautomation.universal_robots.primary_interface.safety_data_package_event_args import SafetyDataPackageEventArgs
from underautomation.universal_robots.primary_interface.tool_communication_info_package_event_args import ToolCommunicationInfoPackageEventArgs
from underautomation.universal_robots.primary_interface.tool_mode_info_package_event_args import ToolModeInfoPackageEventArgs
from underautomation.universal_robots.primary_interface.singularity_info_package_event_args import SingularityInfoPackageEventArgs
from underautomation.universal_robots.primary_interface.program_threads_event_args import ProgramThreadsEventArgs
from underautomation.universal_robots.primary_interface.version_event_args import VersionEventArgs
from underautomation.universal_robots.primary_interface.key_message_event_args import KeyMessageEventArgs
from underautomation.universal_robots.primary_interface.popup_message_event_args import PopupMessageEventArgs
from underautomation.universal_robots.primary_interface.text_message_event_args import TextMessageEventArgs
from underautomation.universal_robots.primary_interface.runtime_exception_message_event_args import RuntimeExceptionMessageEventArgs
from underautomation.universal_robots.primary_interface.global_variables import GlobalVariables
from underautomation.universal_robots.primary_interface.interfaces import Interfaces
from underautomation.universal_robots.status_code import StatusCode
from underautomation.universal_robots.internal.ur_service_base import URServiceBase
from underautomation.universal_robots.common.package_event_args import PackageEventArgs
import clr
import os
clr.AddReference(os.path.realpath(os.path.join(os.path.dirname(__file__), "..", "..", "..", "..", 'lib', 'UnderAutomation.UniversalRobots.dll')))
from UnderAutomation.UniversalRobots.PrimaryInterface.Internal import PrimaryInterfaceClientBase as primary_interface_client_base

class PrimaryInterfaceClientBase(URServiceBase):
	def __init__(self, _internal = 0):
		if(_internal == 0):
			self._instance = primary_interface_client_base()
		else:
			self._instance = _internal
	def robot_mode_data_received(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.RobotModeDataReceived+= lambda sender, e : handler(Wrapper(sender), Wrapper(e))
	def joint_data_received(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.JointDataReceived+= lambda sender, e : handler(Wrapper(sender), Wrapper(e))
	def tool_data_received(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.ToolDataReceived+= lambda sender, e : handler(Wrapper(sender), Wrapper(e))
	def masterboard_data_received(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.MasterboardDataReceived+= lambda sender, e : handler(Wrapper(sender), Wrapper(e))
	def cartesian_info_received(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.CartesianInfoReceived+= lambda sender, e : handler(Wrapper(sender), Wrapper(e))
	def kinematics_info_received(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.KinematicsInfoReceived+= lambda sender, e : handler(Wrapper(sender), Wrapper(e))
	def configuration_data_received(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.ConfigurationDataReceived+= lambda sender, e : handler(Wrapper(sender), Wrapper(e))
	def force_mode_data_received(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.ForceModeDataReceived+= lambda sender, e : handler(Wrapper(sender), Wrapper(e))
	def additional_info_received(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.AdditionalInfoReceived+= lambda sender, e : handler(Wrapper(sender), Wrapper(e))
	def calibration_data_received(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.CalibrationDataReceived+= lambda sender, e : handler(Wrapper(sender), Wrapper(e))
	def safety_data_received(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.SafetyDataReceived+= lambda sender, e : handler(Wrapper(sender), Wrapper(e))
	def tool_communication_info_received(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.ToolCommunicationInfoReceived+= lambda sender, e : handler(Wrapper(sender), Wrapper(e))
	def tool_mode_info_received(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.ToolModeInfoReceived+= lambda sender, e : handler(Wrapper(sender), Wrapper(e))
	def singularity_info_received(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.SingularityInfoReceived+= lambda sender, e : handler(Wrapper(sender), Wrapper(e))
	def package_received(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.PackageReceived+= lambda sender, e : handler(Wrapper(sender), Wrapper(e))
	def program_threads_received(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.ProgramThreadsReceived+= lambda sender, e : handler(Wrapper(sender), Wrapper(e))
	def version_received(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.VersionReceived+= lambda sender, e : handler(Wrapper(sender), Wrapper(e))
	def key_message_received(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.KeyMessageReceived+= lambda sender, e : handler(Wrapper(sender), Wrapper(e))
	def popup_message_received(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.PopupMessageReceived+= lambda sender, e : handler(Wrapper(sender), Wrapper(e))
	def text_message_received(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.TextMessageReceived+= lambda sender, e : handler(Wrapper(sender), Wrapper(e))
	def runtime_exception_message_received(self, handler):
		class Wrapper :
			def __init__(self, _internal):
				self._instance = _internal
		self._instance.RuntimeExceptionMessageReceived+= lambda sender, e : handler(Wrapper(sender), Wrapper(e))
	def disconnect(self) -> None:
		self._instance.Disconnect()
	def send(self, script: str) -> StatusCode:
		return StatusCode(self._instance.Send(script))
	@property
	def robot_mode_data(self) -> RobotModeDataPackageEventArgs:
		return RobotModeDataPackageEventArgs(self._instance.RobotModeData)
	@property
	def joint_data(self) -> JointDataPackageEventArgs:
		return JointDataPackageEventArgs(self._instance.JointData)
	@property
	def tool_data(self) -> ToolDataPackageEventArgs:
		return ToolDataPackageEventArgs(self._instance.ToolData)
	@property
	def masterboard_data(self) -> MasterboardDataPackageEventArgs:
		return MasterboardDataPackageEventArgs(self._instance.MasterboardData)
	@property
	def cartesian_info(self) -> CartesianInfoPackageEventArgs:
		return CartesianInfoPackageEventArgs(self._instance.CartesianInfo)
	@property
	def kinematics_info(self) -> KinematicsInfoPackageEventArgs:
		return KinematicsInfoPackageEventArgs(self._instance.KinematicsInfo)
	@property
	def configuration_data(self) -> ConfigurationDataPackageEventArgs:
		return ConfigurationDataPackageEventArgs(self._instance.ConfigurationData)
	@property
	def force_mode_data(self) -> ForceModeDataPackageEventArgs:
		return ForceModeDataPackageEventArgs(self._instance.ForceModeData)
	@property
	def additional_info(self) -> AdditionalInfoPackageEventArgs:
		return AdditionalInfoPackageEventArgs(self._instance.AdditionalInfo)
	@property
	def calibration_data(self) -> CalibrationDataPackageEventArgs:
		return CalibrationDataPackageEventArgs(self._instance.CalibrationData)
	@property
	def safety_data(self) -> SafetyDataPackageEventArgs:
		return SafetyDataPackageEventArgs(self._instance.SafetyData)
	@property
	def tool_communication_info(self) -> ToolCommunicationInfoPackageEventArgs:
		return ToolCommunicationInfoPackageEventArgs(self._instance.ToolCommunicationInfo)
	@property
	def tool_mode_info(self) -> ToolModeInfoPackageEventArgs:
		return ToolModeInfoPackageEventArgs(self._instance.ToolModeInfo)
	@property
	def singularity_info(self) -> SingularityInfoPackageEventArgs:
		return SingularityInfoPackageEventArgs(self._instance.SingularityInfo)
	@property
	def program_threads(self) -> ProgramThreadsEventArgs:
		return ProgramThreadsEventArgs(self._instance.ProgramThreads)
	@property
	def version(self) -> VersionEventArgs:
		return VersionEventArgs(self._instance.Version)
	@property
	def key_message(self) -> KeyMessageEventArgs:
		return KeyMessageEventArgs(self._instance.KeyMessage)
	@property
	def popup_message(self) -> PopupMessageEventArgs:
		return PopupMessageEventArgs(self._instance.PopupMessage)
	@property
	def text_message(self) -> TextMessageEventArgs:
		return TextMessageEventArgs(self._instance.TextMessage)
	@property
	def runtime_exception_message(self) -> RuntimeExceptionMessageEventArgs:
		return RuntimeExceptionMessageEventArgs(self._instance.RuntimeExceptionMessage)
	@property
	def global_variables(self) -> GlobalVariables:
		return GlobalVariables(self._instance.GlobalVariables)
	@property
	def ip(self) -> str:
		return self._instance.IP
	@property
	def port(self) -> Interfaces:
		return Interfaces(self._instance.Port)
	@property
	def connected(self) -> bool:
		return self._instance.Connected
	@property
	def local_end_point(self) -> typing.Any:
		return self._instance.LocalEndPoint

File: UnderAutomation.UniversalRobots.ObfuscatedSources/PrimaryInterface/Internal/PrimaryInterfaceClientBase.cs
//              WARNING
// This file is an empty shell containing only public C# items.
// The internal code is hidden; to access it, you need to obtain a Source licence of the library.

using Internal;
using System;
using PrimaryInterface;
using Common;
using System.Net;

namespace PrimaryInterface.Internal {

	public class PrimaryInterfaceClientBase : URServiceBase {

		/// <summary>
		/// Robot mode data (Raised every 100ms)
		/// </summary>
		public event EventHandler<RobotModeDataPackageEventArgs> RobotModeDataReceived;

		/// <summary>
		/// Joint data (Raised every 100ms)
		/// </summary>
		public event EventHandler<JointDataPackageEventArgs> JointDataReceived;

		/// <summary>
		/// Tool data (Raised every 100ms)
		/// </summary>
		public event EventHandler<ToolDataPackageEventArgs> ToolDataReceived;

		/// <summary>
		/// Masterboard data (Raised every 100ms)
		/// </summary>
		public event EventHandler<MasterboardDataPackageEventArgs> MasterboardDataReceived;

		/// <summary>
		/// Cartesian inforlation (Raised every 100ms)
		/// </summary>
		public event EventHandler<CartesianInfoPackageEventArgs> CartesianInfoReceived;

		/// <summary>
		/// Kinematics information data (Raised when connection opened)
		/// </summary>
		public event EventHandler<KinematicsInfoPackageEventArgs> KinematicsInfoReceived;

		/// <summary>
		/// Configuration data (Raised when connection opened)
		/// </summary>
		public event EventHandler<ConfigurationDataPackageEventArgs> ConfigurationDataReceived;

		/// <summary>
		/// Force mode data (Raised every 100ms)
		/// </summary>
		public event EventHandler<ForceModeDataPackageEventArgs> ForceModeDataReceived;

		/// <summary>
		/// Additional (Raised every 100ms)
		/// </summary>
		public event EventHandler<AdditionalInfoPackageEventArgs> AdditionalInfoReceived;

		/// <summary>
		/// Calibration data (Raised every 100ms)
		/// </summary>
		public event EventHandler<CalibrationDataPackageEventArgs> CalibrationDataReceived;

		/// <summary>
		/// Safety data (Raised every 100ms)
		/// </summary>
		public event EventHandler<SafetyDataPackageEventArgs> SafetyDataReceived;

		/// <summary>
		/// Tool communication information (Raised every 100ms)
		/// </summary>
		public event EventHandler<ToolCommunicationInfoPackageEventArgs> ToolCommunicationInfoReceived;

		/// <summary>
		/// Tool mode information (Raised every 100ms)
		/// </summary>
		public event EventHandler<ToolModeInfoPackageEventArgs> ToolModeInfoReceived;

		/// <summary>
		/// Singularity information (Raised every 100ms)
		/// </summary>
		public event EventHandler<SingularityInfoPackageEventArgs> SingularityInfoReceived;

		/// <summary>
		/// Generic event raised each time a package is received
		/// </summary>
		public event EventHandler<PackageEventArgs> PackageReceived;

		/// <summary>
		/// Program threads changed
		/// </summary>
		public event EventHandler<ProgramThreadsEventArgs> ProgramThreadsReceived;

		/// <summary>
		/// Robot information and FW version
		/// </summary>
		public event EventHandler<VersionEventArgs> VersionReceived;

		/// <summary>
		/// Internal robot events (such as starting or stopping a program)
		/// </summary>
		public event EventHandler<KeyMessageEventArgs> KeyMessageReceived;

		/// <summary>
		/// Popup message that appears with the Assignment instruction or the URScript popup() function
		/// </summary>
		public event EventHandler<PopupMessageEventArgs> PopupMessageReceived;

		/// <summary>
		/// Log message sent with URScript instruction textmsg()
		/// </summary>
		public event EventHandler<TextMessageEventArgs> TextMessageReceived;

		/// <summary>
		/// Reports an error in the execution of the program
		/// </summary>
		public event EventHandler<RuntimeExceptionMessageEventArgs> RuntimeExceptionMessageReceived;

		/// <summary>
		/// Last Robot mode data received
		/// </summary>
		public RobotModeDataPackageEventArgs RobotModeData { get; }

		/// <summary>
		/// Last joint data received
		/// </summary>
		public JointDataPackageEventArgs JointData { get; }

		/// <summary>
		/// Last tool data received
		/// </summary>
		public ToolDataPackageEventArgs ToolData { get; }

		/// <summary>
		/// Last masterboard data received
		/// </summary>
		public MasterboardDataPackageEventArgs MasterboardData { get; }

		/// <summary>
		/// Last cartesian information received
		/// </summary>
		public CartesianInfoPackageEventArgs CartesianInfo { get; }

		/// <summary>
		/// Last kinematics information received
		/// </summary>
		public KinematicsInfoPackageEventArgs KinematicsInfo { get; }

		/// <summary>
		/// Last configuration data received
		/// </summary>
		public ConfigurationDataPackageEventArgs ConfigurationData { get; }

		/// <summary>
		/// Last force mode data received
		/// </summary>
		public ForceModeDataPackageEventArgs ForceModeData { get; }

		/// <summary>
		/// Last additional information received
		/// </summary>
		public AdditionalInfoPackageEventArgs AdditionalInfo { get; }

		/// <summary>
		/// Last calibration data received
		/// </summary>
		public CalibrationDataPackageEventArgs CalibrationData { get; }

		/// <summary>
		/// Last safety data received
		/// </summary>
		public SafetyDataPackageEventArgs SafetyData { get; }

		/// <summary>
		/// Last tool communication information received
		/// </summary>
		public ToolCommunicationInfoPackageEventArgs ToolCommunicationInfo { get; }

		/// <summary>
		/// Last tool mode information received
		/// </summary>
		public ToolModeInfoPackageEventArgs ToolModeInfo { get; }

		/// <summary>
		/// Last singularity information information received
		/// </summary>
		public SingularityInfoPackageEventArgs SingularityInfo { get; }

		/// <summary>
		/// Last program thread information received
		/// </summary>
		public ProgramThreadsEventArgs ProgramThreads { get; }

		/// <summary>
		/// Version of the robot and FW
		/// </summary>
		public VersionEventArgs Version { get; }

		/// <summary>
		/// Internal robot events (such as starting or stopping a program)
		/// </summary>
		public KeyMessageEventArgs KeyMessage { get; }

		/// <summary>
		/// Popup message that appears with the Assignment instruction or the URScript popup() function
		/// </summary>
		public PopupMessageEventArgs PopupMessage { get; }

		/// <summary>
		/// Log message sent with URScript instruction textmsg()
		/// </summary>
		public TextMessageEventArgs TextMessage { get; }

		/// <summary>
		/// Reports an error in the execution of the program
		/// </summary>
		public RuntimeExceptionMessageEventArgs RuntimeExceptionMessage { get; }

		/// <summary>
		/// List of all variables in current robot program
		/// </summary>
		public GlobalVariables GlobalVariables { get; }

		/// <summary>
		/// Contains methods to send custom URScript to the robot
		/// </summary>
		public PrimaryInterfaceScript Script { get; }

		/// <summary>
		/// Contains methods to send commands to the robot
		/// </summary>
		public PrimaryInterfaceCommands Commands { get; }

		/// <summary>
		/// IP address of the connected robot
		/// </summary>
		public string IP { get; }

		/// <summary>
		/// Interface used for the connected robot
		/// </summary>
		public Interfaces Port { get; }


		protected void InternalConnect(string ip, Interfaces port)
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}

		/// <summary>
		/// Stops data streaming and the possibility to send scripts to the robot.
		/// </summary>
		public void Disconnect()
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}

		/// <summary>
		/// Return True if the connection to the robot is active
		/// </summary>
		public bool Connected { get; }

		/// <summary>
		/// Indicates the current local endpoint (i.e. IP Address) used to communicate with the robot. You can use this IP in your UR script in the function rpc_factory()
		/// </summary>
		public IPEndPoint LocalEndPoint { get; }
	}
}
